pkg/hubble/relay/pool/manager_test.go

```diff
@@ -704,6 +709,176 @@ func TestPeerManager(t *testing.T) {
+									},
+									{
+										Name:    "peererino_cuatro",
+										Address: "192.0.1.4",
+										Type:    peerpb.ChangeNotificationType_PEER_ADDED,
+									},
+									{
+										Name:    "peererino_cinco",
+										Address: "192.0.1.5",
+										Type:    peerpb.ChangeNotificationType_PEER_ADDED,
+									},
+									{
+										Name:    "peererino_seis",
+										Address: "192.0.1.6",
+										Type:    peerpb.ChangeNotificationType_PEER_ADDED,
+									},
+									{
+										Name:    "peererino_siete",
+										Address: "192.0.1.7",
+										Type:    peerpb.ChangeNotificationType_PEER_ADDED,
+									},
+								}
+								return &testutils.FakePeerNotifyClient{
+									OnRecv: func() (*peerpb.ChangeNotification, error) {
+										i++
+										switch {
+										case i >= len(cns):
+											once.Do(func() { close(done) })
+											return nil, io.EOF
+										default:
+											return cns[i], nil
+										}
+									},
+								}, nil
+							},
+							OnClose: func() error {
+								return nil
+							},
+						}, nil
+					}
+				}(),
+			},
+			ccBuilder: FakeClientConnBuilder{
+				OnClientConn: func() onClientConnFunc {
+					var i int
+					return func(target, hostname string) (poolTypes.ClientConn, error) {
+						return testutils.FakeClientConn{
+							OnGetState: func() connectivity.State {
+								states := []connectivity.State{
+									connectivity.Idle,
+									connectivity.Ready,
+									connectivity.Connecting,
+									connectivity.TransientFailure,
+									connectivity.Shutdown,
+								}
+								resultState := states[i%len(states)]
+								i = (i + 1) % 7
+								return resultState
+							},
+							OnClose: func() error {
+								return nil
+							},
+						}, nil
+					}
+				}(),
+			},
+			wantStatus: map[string]uint32{
+				"READY":             2,
+				"CONNECTING":        1,
+				"IDLE":              2,
+				"TRANSIENT_FAILURE": 1,
+				"SHUTDOWN":          1,
+				"NIL_CONNECTION":    0,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			done = make(chan struct{})
+			once = sync.Once{}
+
+			var buf bytes.Buffer
+			formatter := &logrus.TextFormatter{
+				DisableColors:    true,
+				DisableTimestamp: true,
+			}
+			logger := logrus.New()
+			logger.SetOutput(&buf)
+			logger.SetFormatter(formatter)
+			logger.SetLevel(logrus.DebugLevel)
+
+			registry := prometheus.NewPedanticRegistry()
+			options := []Option{
+				WithPeerClientBuilder(tt.pcBuilder),
+				WithClientConnBuilder(tt.ccBuilder),
+				WithLogger(logger),
+				WithConnStatusInterval(2 * time.Second),
+				// set interval large enough not to fire in 3 seconds sleep
+				WithConnCheckInterval(20 * time.Minute),
+			}
+
+			mgr, err := NewPeerManager(registry, options...)
+			assert.NoError(t, err)
+			mgr.Start()
+			// sleep so that metrics will be emitted at least once
+			time.Sleep(3 * time.Second)
```

Comment:
Since you're using testify framework for tests, `assert.Eventually` is IMO a clean approach to what you're suggesting. And I assume that loop would need a timeout anyway, so it fits very well here. Changed the test to use this instead of that pesky sleep.